developing/command-line-tools/HOW2GORELEASER.md

```diff
@@ -0,0 +1,357 @@
+
+```sh
+git remote add origin https://github.com/OWNER/NEWREPO.git
+```
+
+Push history:
+
+```sh
+git push origin .
+```
+
+### References
+
+* [GitHub's documentation](https://help.github.com/en/github/using-git/splitting-a-subfolder-out-into-a-new-repository)
+* [Chris Gilmer’s full instructions](
+https://github.com/chrisgilmerproj/silliness#how-to-break-out-projects)
+
+## Get the project building binaries
+
+This assumes you are creating a binary from a Go project.
+
+### Generate a `go.mod` file
+
+Your project may not have a `go.mod` file already, you can skip ahead if so.
+
+Initializes your `go.mod` file with your current version of Go:
+
+```sh
+go mod init github.com/OWNER/NEWREPO
+```
+
+Fills out your `go.mod` and `go.sum` files and builds your binary for your current OS:
+
+```sh
+go build
+```
+
+You might run into some issues getting this to work outside of a project you may have split this code from. Spend some time getting it to build.
+
+When you feel good about it, add `go.mod` and `go.sum` to your git index with a commit.
+
+[Blog post I used to figure this out](https://medium.com/mindorks/create-projects-independent-of-gopath-using-go-modules-802260cdfb51)
+
+## Add pre-commit hooks
+
+Write a `.pre-commit-config.yaml` file and add it to git with a commit.
+Then install the hooks and run the hooks against all files for the first time.
+
+```sh
+pre-commit install --install-hooks
+pre-commit run --all-files
+```
+
+Fix any issues you find and commit your changes to your repo.
+
+### Example .pre-commit-config.yaml and .markdownlintrc
+
+The following example `.pre-commit-config.yaml` does some nice things for a basic go project.
+
+```yml
+repos:
+  - repo: git://github.com/golangci/golangci-lint
+    rev: v1.21.0
+    hooks:
+      - id: golangci-lint
+
+  - repo: git://github.com/pre-commit/pre-commit-hooks
+    rev: v2.4.0
+    hooks:
+      - id: check-json
+      - id: check-merge-conflict
+      - id: check-yaml
+      - id: detect-private-key
+      - id: pretty-format-json
+        args:
+          - --autofix
+      - id: trailing-whitespace
+
+  - repo: git://github.com/igorshubovych/markdownlint-cli
+    rev: v0.21.0
+    hooks:
+      - id: markdownlint
+```
+
+This example `.markdownlintrc` will work with the above `.pre-commit-config.yaml` example.
+This ignores some stylistic but annoying triggers.
+
+```json
+{
+  "default": true,
+  "first-header-h1": false,
+  "first-line-h1": false,
+  "line_length": false,
+  "no-multiple-blanks": false,
+  "fenced-code-language": false
+}
+```
+
+## Create Docker configuration
+
+Suggested but optional. This is just another distribution method.
+
+Create a `Dockerfile` at the root of your repo and commit it to git.
+
+### Example Dockerfile
+
+This example is a very basic Dockerfile that works with goreleaser with a tool/binary name of `binaryname`.
+
+Goreleaser will use the binary it determines is correct to copy into the container.
+
+```docker
+FROM alpine:3
```

Comment:
Would it make sense to limit this to something like scratch or one of the distroless containers?

Reply:
Maybe? probably? I'm not familiar with what are the tradeoffs.